android/quest/src/test/java/org/smartregister/fhircore/quest/ui/pin/PinViewModelTest.kt

```diff
@@ -58,14 +61,28 @@ class PinViewModelTest : RobolectricTest() {
   @Test
   fun testSetPinUiState() {
     val context = ApplicationProvider.getApplicationContext<Application>()
+    val applicationConfiguration = mockk<ApplicationConfiguration>()
+    val pinLoginState: MutableState<PinUiState> =
+      mutableStateOf(
+        PinUiState(
+          currentUserPin = "1245",
+          message = "VHT will use this PIN to login",
+          appName = "demo",
+          setupPin = true,
+          pinLength = 4, // set pinLength to a value greater than 0
+          showLogo = false
+        )
+      )
     every { secureSharedPreference.retrieveSessionPin() } returns "1245"
     every { secureSharedPreference.retrieveSessionUsername() } returns "demo"
-    pinViewModel.setPinUiState(true, context)
-    val pinUiState = pinViewModel.pinUiState.value
-    Assert.assertEquals("1245", pinUiState.currentUserPin)
-    Assert.assertTrue(pinUiState.setupPin)
+    pinViewModel.pinUiState.value = pinLoginState.value
+    val expectedMessage = "VHT will use this PIN to login"
```

Comment:
to reduce duplication and guarantee equality, please move this `val` above the definition of PinUiState and then pass it as an argument on initialization